File: CS-Examples/09_Charts/ChartToImage.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using System.Drawing.Imaging;
using Spire.Xls;

namespace ChartToImage
{
	public partial class Form1 :Form
	{
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a workbook
			Workbook workbook = new Workbook();

            //Load file from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ChartToImage.xlsx");

            //Save chart as image
            Image image= workbook.SaveChartAsImage(workbook.Worksheets[0], 0);
            image.Save("Output.png",ImageFormat.Png);
            
            //////////////////Use the following code for netstandard dlls/////////////////////////
			/*
            Stream image = workbook.SaveChartAsImage(workbook.Worksheets[0], 0);
            string filename = String.Format("ChartToImage.png");
            FileStream fileStream = new FileStream(filename, FileMode.Create, FileAccess.Write);
            image.CopyTo(fileStream, 100);
            fileStream.Flush();
            fileStream.Close();
            image.Close();
			*/
			
            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the file
            ExcelDocViewer("Output.png");
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: CS-Examples/09_Charts/ChartToImage.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing.Imaging
Imports Spire.Xls

Namespace ChartToImage
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file "ChartToImage.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ChartToImage.xlsx")

            ' Save the chart from the first worksheet of the workbook as an image and assign it to the 'image' variable
            Dim image As Image = workbook.SaveChartAsImage(workbook.Worksheets(0), 0)

            ' Save the image to a file named "Output.png" in PNG format
            image.Save("Output.png", ImageFormat.Png)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the file
            ExcelDocViewer("Output.png")
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
